src/main/java/trinity/play2learn/backend/admin/year/services/YearRegisterService.java

```diff
@@ -0,0 +1,52 @@
+package trinity.play2learn.backend.admin.year.services;
+
+import org.springframework.stereotype.Service;
+import org.springframework.validation.BindingResult;
+
+import trinity.play2learn.backend.admin.year.dtos.YearRequestDto;
+import trinity.play2learn.backend.admin.year.dtos.YearResponseDto;
+import trinity.play2learn.backend.admin.year.mappers.YearMapper;
+import trinity.play2learn.backend.admin.year.models.Year;
+import trinity.play2learn.backend.admin.year.repositories.IYearRepository;
+import trinity.play2learn.backend.admin.year.services.interfaces.IYearExistService;
+import trinity.play2learn.backend.admin.year.services.interfaces.IYearRegisterService;
+import trinity.play2learn.backend.configs.exceptions.BadRequestException;
+
+/**
+ * Servicio para la lógica de negocio relacionada con el caso de uso 7, registrar un año académico.
+ * Encapsula validaciones y operaciones sobre el modelo Year.
+ */
+@Service
+public class YearRegisterService implements IYearRegisterService {
+
+    private IYearExistService  yearExistService;
+    private IYearRepository yearRepository;
+
+    public YearRegisterService(IYearExistService yearExistService, IYearRepository yearRepository) {
+        this.yearExistService = yearExistService;
+        this.yearRepository = yearRepository;
+    }
```

Comment:
Es buena práctica que los beans sean finales para evitar asignaciones posteriores (que es muy raro que aparezcan, pero por las dudas).

Por otro lado, ya que usás lombok, podrías ahorrarte el constructor

```suggestion
@Service
@AllArgsConstructor
public class YearRegisterService implements IYearRegisterService {

    private final IYearExistService yearExistService;
    private final IYearRepository yearRepository;
```